### ISSUE TITLE
Explicit Api Key Configuration No Longer Works

### ISSUE THREAD

**Issue by rdegges** [stormpath/express-stormpath#272]
It appears that explicit API key configuration no longer works...

First off, ensure you have no `~/.stormpath/apiKey.properties` file present, or a local `apiKey.properties` file.

Secondly, ensure you have no Stormpath environment variables set.

Next, run the following program:

```javascript
ar express = require('express');
var stormpath = require('express-stormpath');

var app = express();

app.use(stormpath.init(app, {
  application: {
    href: 'https://api.stormpath.com/v1/applications/xxx'
  },
  // these settings below are NOT respected
  client: {
    apiKey: {
      id: 'xxx',
      secret: 'yyy'
    }
  },
  website: true
}));

app.on('stormpath.ready', () => {
  app.listen(process.env.PORT || 3000);
});
```

**Comment by robertjd** [comment]
I can confirm this problem, with your example.  This does not appear to be a problem with the Node SDK, but rather with the client configuration strategy in `lib/client.js` in this module.  @typerandom is the best person to provide insight.

**Comment by typerandom** [comment]
I can confirm this too. And looking into it. I think we need to have two things:

1) Ability to easier debug what happens when configuration is being loaded.
2) Tests for configuration in `express-stormpath`. I thought we had this, but it's obviously been overlooked.

**Comment by rdegges** [comment]
We should have some full IT test cases in express-stormpath for verifying all of these use cases, I think. It might be a pain to write initially, but should be really helpful in preventing regressions.

**Comment by typerandom** [comment]
@rdegges I agree with that we should add tests in order to verify all of these cases.

I'm actually unable to replicate this issue at the moment. Are you still able to do that with the latest express-stormpath version?

@timothyej Could you try and see if you can replicate this issue?

**Comment by robertjd** [comment]
This is fixed in >= 2.3.6, where we updated to stormpath-config@0.0.18

FYI, the `client` key is NOT necessary - `apiKey` can be at the root of the configuration object and this is [how we example the usage of client options](http://docs.stormpath.com/nodejs/express/latest/configuration.html#stormpath-client-options).  The fact that `config` works is due to some legacy logic we have in the config parser for supporting old configuration input from old SDK versions.

Closing.
